src/Util/Visitor/ReturnNodeVisitor.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace BetterReflection\Util\Visitor;
+
+use PhpParser\NodeVisitorAbstract;
+use PhpParser\Node;
+
+class ReturnNodeVisitor extends NodeVisitorAbstract
+{
+    /**
+     * @var Node\Stmt\Return_[]
+     */
+    private $returnNodes = [];
+
+    private $scopeDepth = 1;
+
+    public function enterNode(Node $node)
+    {
+        if ($this->isScopeChangingNode($node)) {
+            $this->scopeDepth++;
+        }
```

Comment:
An alternative to managing the scope depth would be to return `NodeTraverser::DONT_TRAVERSE_CHILDREN` here.